mlx/ops.cpp

```diff
@@ -3057,4 +3057,110 @@ array inner(const array& a, const array& b, StreamOrDevice s /* = {} */) {
   return tensordot(a, b, {{-1}, {-1}}, s);
 }
 
+/** Compute the matrix product of A and B and add C to the result */
+array addmm(
+    const array& in_c,
+    const array& in_a,
+    const array& in_b,
```

Comment:
I would just make this pass by value instead. Since we always copy them anyway. I think it would be good to start moving to pass by value for these cases as it gives us the opportunity to use move semantics

Reply:
Got it 
Should I update the `matmul` op as well ?

Reply:
Scratch that, we need the originals for padding dims in gemv cases there